### PR TITLE
Only compile CLVM if source newer than hex

### PR DESCRIPTION
**Pull request by Quexington**
Running tests is really slow right now because _every_ clvm file is recompiled on _every_ test run.  This reverts to the old behavior of only recompiling if the source is newer than the hex.  The behavior remains guarded behind an environment variable or running during pytest.

### CODE REVIEW

**Review by emlowe — APPROVED** [review]
Looks good to me

**Comment by arvidn** [comment]
we check in the compiled code too, right? so this test (as far as I understand it) is mostly to ensure the checked-in compiled programs match the corresponding source code.

Is looking at timestamps a good proxy for these to possibly be out of sync?
I suppose it would detect a commit *failing* to compile, but not to detect whether it's *correctly* compiled.

**Comment by arvidn** [comment]
I think @AmineKhaldi and @prozacchiwawa should see this before we land

**Comment by prozacchiwawa** [comment]
i don't think this breaks anything.  will approve if wanted.

**Comment by altendky** [comment]
It doesn't particularly matter, but I'm curious when this behavior got lost.  I don't think I removed it in https://github.com/Chia-Network/chia-blockchain/commit/a20ee08fce6d608bf0f886b8edba35031dabb1c1.  I looked a bit farther, just real quickly, and didn't see anything else.

**Comment by Quexington** [comment]
> It doesn't particularly matter, but I'm curious when this behavior got lost. I don't think I removed it in [a20ee08](https://github.com/Chia-Network/chia-blockchain/commit/a20ee08fce6d608bf0f886b8edba35031dabb1c1). I looked a bit farther, just real quickly, and didn't see anything else.

The behavior used to be part of the python clvm_tools so maybe switching to clvm_tools_rs at some point changed this behavior.

**Pull request merged**
